### DISCUSSION TITLE
Neovim: next suggestion

### DISCUSSION THREAD

**Discussion by coltonmorris** [community/community#6862] [Copilot Conversations] [answered]
Hey, really good work on the neovim plugin! It's working well so far.

If it's not already in the roadmap, cycling through suggestions would be nice.

Thanks

**Comment by TarunavBA** [comment]
Hey @coltonmorris NeoVim is already supported. Take a look at https://copilot.github.com/ in the availability section nearly at the end of the page.

**Reply by AceofSpades5757** [reply]
Right, but it doesn't seem to have support for alternate solutions, such as seen [here](https://github.com/github/copilot-docs/blob/main/docs/visualstudiocode/gettingstarted.md#3-choosing-alternate-suggestions) for VS Code, or listing additional suggestions as seen [here](https://github.com/github/copilot-docs/blob/main/docs/visualstudiocode/gettingstarted.md#4-getting-more-suggestions), also for VS Code.

I think something like these features are being requested. I would also like to see something like this added.

**Reply by IvRRimum** [reply]
+1

**Comment by tpope** [comment, marked as answer]
It's ~~on the roadmap~~ now supported!

**Comment by metal3d** [comment]
At this time, I change comments on my code to try to "help" the IA to give better suggestions (yeah... dirty workaround 🥲 )

**Comment by bneradt** [comment]
Can I have some clarification concerning Neovim support of Copilot next and previous suggestions? In the `:help Copilot` file, it has a mapping section which shows this:

```
                                                copilot-i_ALT-]                                                                                                                                                                                    
<M-]>                   Cycle to the next suggestion, if one is available.                                                                                                                                                                         
<Plug>(copilot-next)                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                   
                                                copilot-i_ALT-[                                                                                                                                                                                    
<M-[>                   Cycle to the previous suggestion.                                                                                                                                                                                          
<Plug>(copilot-previous)                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                   
```

I'm having trouble getting that to work, but I figured that was perhaps because of conflicts with my other plugins. I've tried switching the mapping and I haven't succeeded yet. I'm on a Mac, FWIW.

Can someone confirm: am I on a wild goose chase trying to get next/previous Copilot suggestions to work? Or should I keep digging? This thread seems to indicate that it is on the roadmap and does not currently work, while the `:help Copilot` documentation would seem to imply that I should be able to get it to work.

Either way, thanks for your work on this excellent project.

Brian

**Reply by tpope** [reply]
Mac terminals typically don't provide access to alt/meta out of the box. On Terminal.app you can go into Preferences >> Profiles >> Keyboard and check "Use option as meta key".  On iTerm2 it's at Preferences >> Profiles >> Keys: Choose "Esc+" for either the left or right option key.

You can also create your own maps to `<Plug>(copilot-next)` and `<Plug>(copilot-previous)`.

**Reply by bneradt** [reply]
> Mac terminals typically don't provide access to alt/meta out of the box. On Terminal.app you can go into Preferences >> Profiles >> Keyboard and check "Use option as meta key". On iTerm2 it's at Preferences >> Profiles >> Keys: Choose "Esc+" for either the left or right option key.
> 
> You can also create your own maps to `<Plug>(copilot-next)` and `<Plug>(copilot-previous)`.

"Use option as meta key" in Terminal.app worked. Thank you for the quick reply!

And, to explicitly answer my question, I suppose this thread is outdated and next and previous Copilot suggestions are implemented in Neovim.

**Reply by bneradt** [reply]
Oh, I see, you updated the text demonstrating that it is implemented. Thank you @tpope !

**Reply by foxt451** [reply]
Just did so on mac, and instead of outputting ' " it now prints [ or ], waits a sec or two and exits insert mode. When I try pressing `i` again, it prints error: `No identifier under cursor`, or just ignores one next key I press. In `cat` pressing opt-] outputs `^]]`. What could I do?

**Reply by tpope** [reply]
@fox451 guessing you're in Vim, not Neovim. If so, alt/meta isn't properly supported there. Your best options are to pick another key to map `<Plug>(copilot-next)` to, or switch to Neovim.

**Comment by gwd999** [comment]
after I switched to a paid plan my neovim copilot is no longer working - constantly complaining about not fully configured or incomplete setup. But the Action to "Complete Settings" will just take me to the rather slim Copilot Settings on my Profile; anyone has an idea what Settings neovim Copilot might be missing?

**Reply by tpope** [reply]
Make sure you press "Save and get started" if you haven't already. If that doesn't work, I would [contact GitHub support](https://support.github.com/), as it sounds similar to [an issue experienced by a handful of VS Code issues](https://github.com/orgs/github-community/discussions/19105#discussioncomment-3002963).
